src/main/java/com/rcv/ElectionConfig.java

```diff
@@ -59,23 +62,119 @@ boolean validate() {
     boolean isValid = true;
 
     Logger.executionLog(Level.INFO, "Validating config...");
-    // TODO: reorder all checks so they go in same order as fields in config GUI
-    // TODO: need to add checks that all required String fields !.equals("")
 
-    if (getNumDeclaredCandidates() == 0) {
+    if (getContestName() == null || getContestName().isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Contest name is required.");
+    }
+
+    if (getOutputDirectory() == null || getOutputDirectory().isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Output directory is required.");
+    }
+
+    if (rawConfig.cvrFileSources == null || rawConfig.cvrFileSources.isEmpty()) {
+      isValid = false;
+      Logger.executionLog(Level.SEVERE, "Config doesn't contain any CVR files.");
+    } else {
```

Comment:
Could you factor this out into a helper? `validate` is getting really long.

Reply:
If the time comes to break this up let's try and keep the UI model and group things by tab (only if it makes sense in the code of course)